skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/aibot/PainterResource.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2024 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.tooling.aibot
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.vector.rememberVectorPainter
+import androidx.compose.ui.platform.LocalDensity
+import org.jetbrains.compose.resources.ExperimentalResourceApi
+import org.jetbrains.compose.resources.decodeToImageBitmap
+import org.jetbrains.compose.resources.decodeToImageVector
+import org.jetbrains.compose.resources.decodeToSvgPainter
+
+// Migration snippet copied from https://github.com/JetBrains/compose-multiplatform-core/pull/1457
+// To resolve deprecated painterResource function from upgrading compose-jb
+class PainterResource {
```

Comment:
This should be an `object`, otherwise you're gonna create a new instance every time. In general - if the class has no internal state, you should default to using top-level functions instead. Maaaaybe an `object` class if you really need the namespacing.

Reply:
Thanks for the tip, I changed it here: 35b22fcc01010d9856c5c824f73034297330f157